controllers/provider-aws/pkg/infrastructure/actuator.go

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infrastructure
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	gardenerLogger "github.com/gardener/gardener/pkg/logger"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/tools/record"
+
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/aws/v1alpha1"
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/aws"
+	"github.com/gardener/gardener-extensions/pkg/controller/infrastructure"
+	"github.com/gardener/gardener/pkg/utils/flow"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/chartrenderer"
+	"github.com/gardener/gardener/pkg/operation/terraformer"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"github.com/go-logr/logr"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	loggerName    = "infrastructure-aws-actuator"
+	accessKeysMap = map[string]string{
+		"ACCESS_KEY_ID":     aws.AccessKeyID,
+		"SECRET_ACCESS_KEY": aws.SecretAccessKey,
+	}
+)
+
+const (
+	eventReasonDestruction = "Destroying Infrastructure"
```

Comment:
Usually, events are one-word identifiers, e.g., in this case "InfrastructureDestruction".
Also, they are defined as constants in the API packages themselves, so if you really want to use them (although, as discussed, I still don't get it why we want it because nobody will evaluate/read them), then put them into `pkg/apis/extensions/v1alpha1/types_infrastructure.go` in Gardener (so that other extensions implementing the resource can also use it).